src/main/kotlin/org/rust/ide/settings/RsCodeInsightSettings.kt

```diff
@@ -5,28 +5,67 @@
 
 package org.rust.ide.settings
 
+import com.intellij.openapi.Disposable
 import com.intellij.openapi.components.PersistentStateComponent
 import com.intellij.openapi.components.State
 import com.intellij.openapi.components.Storage
 import com.intellij.openapi.components.service
 import com.intellij.util.xmlb.XmlSerializerUtil
+import org.rust.openapiext.isUnitTestMode
 
 @State(name = "RsCodeInsightSettings", storages = [Storage("rust.xml")])
-class RsCodeInsightSettings : PersistentStateComponent<RsCodeInsightSettings> {
+class RsCodeInsightSettings : PersistentStateComponent<RsCodeInsightSettings>, Disposable {
 
     var showImportPopup: Boolean = false
     var importOutOfScopeItems: Boolean = true
     var suggestOutOfScopeItems: Boolean = true
     var addUnambiguousImportsOnTheFly: Boolean = false
     var importOnPaste: Boolean = false
+    private var excludedPaths: Array<ExcludedPath>? = null
+
+    fun getExcludedPaths(): Array<ExcludedPath> = excludedPaths ?: DEFAULT_EXCLUDED_PATHS
+
+    fun setExcludedPaths(value: Array<ExcludedPath>) {
+        excludedPaths = if (DEFAULT_EXCLUDED_PATHS.contentEquals(value)) null else value
+    }
 
     override fun getState(): RsCodeInsightSettings = this
 
     override fun loadState(state: RsCodeInsightSettings) {
         XmlSerializerUtil.copyBean(state, this)
     }
 
+    override fun dispose() {}
+
     companion object {
         fun getInstance(): RsCodeInsightSettings = service()
+
+        private val TEST_EXCLUDED_PATHS: Array<ExcludedPath> = arrayOf(
+            ExcludedPath("crate::excluded_mod::*"),
+            ExcludedPath("crate::mod1::excluded_func"),
+            ExcludedPath("crate::mod1::ExcludedTrait", ExclusionType.Methods),
+        )
```

Comment:
I'd prefer these path to be explicitly visible within every test, e.g. with an annotation `@WithExcludedPaths(...)` Such an annotation can be read and applied to `RsCodeInsightSettings` in `doTest()` (see e.g. `RsTraitMethodCompletionTest.doTest()`)